Kosystem.Shared/PersonModel.cs

```diff
@@ -1,19 +1,85 @@
-﻿using System;
+using System;
 
 namespace Kosystem.Shared
 {
-    public class PersonModel
+    public readonly struct PersonModel : IEquatable<PersonModel>
     {
-        public int RoomId { get; set; }
+        public int Id { get; }
 
-        public string Name { get; set; }
+        public int RoomId { get; }
 
-        public DateTime? EnqueuedAt { get; set; }
+        public string Name { get; }
 
-        public PersonModel(int roomId, string name)
+        public DateTime? EnqueuedAt { get; }
+
+        public bool IsInRoom => RoomId > 0;
+
+        public bool IsEnqueued => EnqueuedAt.HasValue;
+
+        public PersonModel(int id, string name)
         {
-            RoomId = roomId;
+            Id = id;
+            Name = name;
+            RoomId = -1;
+            EnqueuedAt = default;
+        }
+
+        public PersonModel(int id, string name, int roomId, DateTime? enqueuedAt)
+        {
+            Id = id;
             Name = name;
+            RoomId = roomId;
+            EnqueuedAt = enqueuedAt;
+        }
+
+        public PersonModel WithoutRoom()
+        {
+            return new PersonModel(Id, Name);
+        }
+
+        public PersonModel WithRoom(int roomId)
+        {
+            return new PersonModel(Id, Name, roomId, EnqueuedAt);
+        }
+
+        public PersonModel AsEnqueued(DateTime enqueuedAt)
+        {
+            return new PersonModel(Id, Name, RoomId, enqueuedAt);
+        }
+
+        public PersonModel AsDequeued()
+        {
+            return new PersonModel(Id, Name, RoomId, null);
+        }
+
+        public PersonModel Update(UpdatePersonModel update)
```

Comment:
Codacy found an issue: [Rename the parameter 'update' so that it does not duplicate the method name.](https://app.codacy.com/gh/HelloWorldSweden/Kosystem/pullRequest?prid=6010716)